components/automate-gateway/gateway/middleware/auth_interceptor.go

```diff
@@ -41,18 +41,16 @@ type SwitchingAuthorizationHandler interface {
 	GRPCAuthorizationHandler
 	SwitchingFilterHandler
 	IsAuthorized(ctx context.Context, subjects []string,
-		resourceV1, actionV1, resourceV2, actionV2 string, projects []string) (AnnotatedAuthorizationResponse, error)
+		resource, action string, projects []string) (AnnotatedAuthorizationResponse, error)
 }
 
 type SwitchingFilterHandler interface {
```

Comment:
Though I stripped out the v1/v2 distinction, it seems that there are further simplifications to be done: removing this SwitchingFilterHandler and the SwitchingAuthorizationHandler, for instance, since they are no longer "switching".

But I feel like I have stared at them long enough for now; if you view this file (auth_interceptor.go) in its entirety, you will begin to the see the... entanglements that would need to be unravelled.